lib/src/connection.rs

```diff
@@ -255,8 +311,14 @@ impl ConnectionInfo {
             },
             port,
             encryption,
+            client_certificate: None
         })
     }
+
+    pub fn with_client_certificate(&mut self, certificate: &ClientCertificate) -> &Self {
```

Comment:
There doesn't seem to be a sensible path where we have a ConnectionInfo without also having an `Option<&ClientCertificate>`, so I would add this to `new` directly and remove this with_ method.